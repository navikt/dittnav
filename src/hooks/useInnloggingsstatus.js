import { useQuery } from 'react-query';
import { fetchInnloggingsstatus } from '../Api';
import useStore from './useStore';

const useInnloggingsstatus = () => (
  [useQuery('innloggingsstatus', fetchInnloggingsstatus, { onError: useStore().setError })]
);

export default useInnloggingsstatus;
