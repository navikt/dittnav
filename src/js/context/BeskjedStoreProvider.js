import React, { createContext, useReducer } from 'react';
import { arrayOf, node } from 'prop-types';
import Api from '../Api';
import {
  ADD_BESKJEDER,
  ADD_INAKTIVE_BESKJEDER,
  ADD_INAKTIV_BESKJED, REMOVE_BESKJED,
} from '../types/Actions';
import BeskjedType from '../types/BeskjedType';

const initialState = (beskjeder, inaktiveBeskjeder) => ({
  beskjeder,
  inaktiveBeskjeder,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BESKJEDER:
      return {
        ...state,
        beskjeder: action.payload,
      };
    case ADD_INAKTIVE_BESKJEDER:
      return {
        ...state,
        inaktiveBeskjeder: action.payload,
      };
    case REMOVE_BESKJED:
      Api.postDone({
        eventId: action.payload.eventId,
        uid: action.payload.uid,
      });
      return {
        ...state,
        beskjeder: state.beskjeder.filter(b => action.payload.uid !== b.uid),
      };
    case ADD_INAKTIV_BESKJED:
      return {
        ...state,
        inaktiveBeskjeder: [...state.inaktiveBeskjeder, action.payload],
      };
    default:
      return state;
  }
};

export const BeskjedStoreContext = createContext(null);

const BeskjedStoreProvider = ({ beskjeder, inaktiveBeskjeder, children }) => {
  const [state, dispatch] = useReducer(reducer, initialState(beskjeder, inaktiveBeskjeder));
  const value = { state, dispatch };

  return (
    <BeskjedStoreContext.Provider value={value}>
      {children}
    </BeskjedStoreContext.Provider>
  );
};

BeskjedStoreProvider.propTypes = {
  beskjeder: arrayOf(BeskjedType),
  inaktiveBeskjeder: arrayOf(BeskjedType),
  children: node.isRequired,
};

BeskjedStoreProvider.defaultProps = {
  beskjeder: null,
  inaktiveBeskjeder: null,
};

export default BeskjedStoreProvider;
