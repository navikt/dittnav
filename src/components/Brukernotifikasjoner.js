import React from 'react';
import { bool } from 'prop-types';
import Beskjed from './brukernotifikasjoner/Beskjed';
import Oppgave from './brukernotifikasjoner/Oppgave';
import Innboks from './brukernotifikasjoner/Innboks';
import InnloggingsModal from './common/InnloggingsModal';
import useStore from '../hooks/useStore';
import { byEventTidspunkt } from '../utils/datoUtils';
import useInnloggingsstatus from '../hooks/useInnloggingsstatus';
import BeskjedType from '../types/BeskjedType';
import OppgaveType from '../types/OppgaveType';
import InnboksType from '../types/InnboksType';

const Brukernotifikasjoner = ({ beskjeder, oppgaver, innbokser, erAktiv, erInaktiv }) => {
  const [{ data: innloggingsstatus, isSuccess }] = useInnloggingsstatus();

  if (useStore().state.visInnloggingsModal) {
    return (
      <InnloggingsModal isOpen onClick={() => null} />
    );
  }

  if (!isSuccess) {
    return null;
  }

  return (
    <>
      {oppgaver && innloggingsstatus && oppgaver.content.sort(byEventTidspunkt)
        .map(o => (
          <Oppgave
            key={o.eventId}
            oppgave={o}
            innloggingsstatus={innloggingsstatus.content}
          />
        ))}
      {beskjeder && beskjeder.content && innloggingsstatus && beskjeder.content.sort(byEventTidspunkt)
        .map(b => (
          <Beskjed
            key={b.uid}
            beskjed={b}
            beskjeder={b}
            innloggingsstatus={innloggingsstatus.content}
            erAktiv={erAktiv}
            erInaktiv={erInaktiv}
          />
        ))}
      {innbokser && innloggingsstatus && (
        <Innboks innbokser={innbokser && innbokser.content} innloggingsstatus={innloggingsstatus.content} />
      )}
    </>
  );
};

Brukernotifikasjoner.propTypes = {
  beskjeder: BeskjedType,
  oppgaver: OppgaveType,
  innbokser: InnboksType,
  erAktiv: bool,
  erInaktiv: bool,
};

Brukernotifikasjoner.defaultProps = {
  beskjeder: null,
  oppgaver: null,
  innbokser: null,
  erAktiv: false,
  erInaktiv: false,
};

export default Brukernotifikasjoner;
