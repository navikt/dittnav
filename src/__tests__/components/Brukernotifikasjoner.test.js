import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import wrapIntl from 'js/IntlTestHelper';
import sinon from 'sinon';
import Brukernotifikasjoner from 'js/components/Brukernotifikasjoner';
import BeskjedStoreProvider from 'js/context/StoreProvider';
import * as useInnloggingsstatus from '../../js/hooks/useInnloggingsstatus';
const ReactTestRenderer = require('react-test-renderer');

jest.mock('react-ga');
let sandbox = null;

beforeEach(() => {
  sandbox = sinon.createSandbox();
});

afterEach(() => {
  useInnloggingsstatus.default.restore();
});

const beskjeder = {
  data: {
    content: [
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '1784093204636',
        uid: '934de6ce-f94f-47de-84d2-639ac2674627',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '1584093204636',
        uid: '9346e6ce-f94f-47de-84g2-639ac2674627',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
    ],
  },
};

const oppgaver = {
  data: {
    content: [
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '1582093204636',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '1587093204636',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
    ],
  },
};

const innbokser = {
  data: {
    content: [
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '2584093204636',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
      {
        eventTidspunkt: '2020-03-13T08:53:24.636Z',
        eventId: '6584093204636',
        tekst: 'Du har en sykemelding som må godkjennes',
        link: 'https://enNyLenke',
        produsent: 'dittnav',
        sistOppdatert: '2020-03-13T08:53:25.002983Z',
        sikkerhetsnivaa: 4,
      },
    ],
  },
};

const innloggingsstatus = {
  securityLevel: '4',
};

test('Brukernotifikasjoner empty', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: null },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner oppgaver={null} innbokser={null} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));
  expect(component.toJSON()).toMatchSnapshot();
});

test('Brukernotifikasjoner with beskjeder', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: innloggingsstatus },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner beskjeder={beskjeder} oppgaver={null} innbokser={null} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));
  expect(component.toJSON()).toMatchSnapshot();
});


test('Brukernotifikasjoner with oppgaver', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: innloggingsstatus },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner oppgaver={oppgaver} innbokser={null} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('Brukernotifikasjoner with innbokser', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: innloggingsstatus },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner beskjeder={null} oppgaver={null} innbokser={innbokser} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('Brukernotifikasjoner with beskjeder, oppgaver and innbokser', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: innloggingsstatus },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner beskjeder={beskjeder} oppgaver={oppgaver} innbokser={innbokser} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('Brukernotifikasjoner with beskjeder, oppgaver and innbokser, but no innlogging', () => {
  sandbox.stub(useInnloggingsstatus, 'default')
    .returns([{
      data: { content: null },
      isLoading: false,
      isSuccess: false,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <MemoryRouter initialEntries={['/person/dittnav']}>
      <BeskjedStoreProvider>
        <Brukernotifikasjoner beskjeder={beskjeder} oppgaver={oppgaver} innbokser={innbokser} />
      </BeskjedStoreProvider>
    </MemoryRouter>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});
