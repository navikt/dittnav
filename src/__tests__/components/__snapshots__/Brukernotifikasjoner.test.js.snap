// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brukernotifikasjoner empty 1`] = `null`;

exports[`Brukernotifikasjoner with beskjeder 1`] = `null`;

exports[`Brukernotifikasjoner with beskjeder, oppgaver and innbokser 1`] = `
Array [
  <a
    className="lenkepanel oppgave lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -429.000000)"
            >
              <g
                transform="translate(382.000000, 429.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M35.5,12 C35.7760001,12 36,12.224 36,12.5 L36,35.5 C36,35.7760001 35.7760001,36 35.5,36 L12.5,36 C12.224,36 12,35.7760001 12,35.5 L12,12.5 C12,12.224 12.224,12 12.5,12 L35.5,12 Z M35,13 L13,13 L13,35 L35,35 L35,13 Z M22.853,25.1470001 C23.048,25.3420001 23.048,25.6590001 22.853,25.8540001 L17.853,30.8540001 C17.756,30.9510001 17.628,31.0000001 17.5,31.0000001 C17.372,31.0000001 17.244,30.9510001 17.146,30.8540001 L15.146,28.8540001 C14.951,28.6590001 14.951,28.3420001 15.146,28.1470001 C15.341,27.9520001 15.658,27.9520001 15.853,28.1470001 L17.5,29.7930001 L22.146,25.1470001 C22.341,24.9520001 22.658,24.9520001 22.853,25.1470001 Z M32.5,29.0000001 C32.7760001,29.0000001 33,29.2240001 33,29.5000001 C33,29.7760001 32.7760001,30.0000001 32.5,30.0000001 L24.5,30.0000001 C24.2240001,30.0000001 24,29.7760001 24,29.5000001 C24,29.2240001 24.2240001,29.0000001 24.5,29.0000001 L32.5,29.0000001 Z M22.853,16.147 C23.048,16.342 23.048,16.659 22.853,16.854 L17.853,21.854 C17.756,21.951 17.628,22 17.5,22 C17.372,22 17.244,21.951 17.146,21.854 L15.146,19.854 C14.951,19.659 14.951,19.342 15.146,19.147 C15.341,18.952 15.658,18.952 15.853,19.147 L17.5,20.793 L22.146,16.147 C22.341,15.952 22.658,15.952 22.853,16.147 Z M32.5,20 C32.7760001,20 33,20.224 33,20.5 C33,20.776 32.7760001,21 32.5,21 L24.5,21 C24.2240001,21 24,20.776 24,20.5 C24,20.224 24.2240001,20 24.5,20 L32.5,20 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
  <a
    className="lenkepanel oppgave lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -429.000000)"
            >
              <g
                transform="translate(382.000000, 429.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M35.5,12 C35.7760001,12 36,12.224 36,12.5 L36,35.5 C36,35.7760001 35.7760001,36 35.5,36 L12.5,36 C12.224,36 12,35.7760001 12,35.5 L12,12.5 C12,12.224 12.224,12 12.5,12 L35.5,12 Z M35,13 L13,13 L13,35 L35,35 L35,13 Z M22.853,25.1470001 C23.048,25.3420001 23.048,25.6590001 22.853,25.8540001 L17.853,30.8540001 C17.756,30.9510001 17.628,31.0000001 17.5,31.0000001 C17.372,31.0000001 17.244,30.9510001 17.146,30.8540001 L15.146,28.8540001 C14.951,28.6590001 14.951,28.3420001 15.146,28.1470001 C15.341,27.9520001 15.658,27.9520001 15.853,28.1470001 L17.5,29.7930001 L22.146,25.1470001 C22.341,24.9520001 22.658,24.9520001 22.853,25.1470001 Z M32.5,29.0000001 C32.7760001,29.0000001 33,29.2240001 33,29.5000001 C33,29.7760001 32.7760001,30.0000001 32.5,30.0000001 L24.5,30.0000001 C24.2240001,30.0000001 24,29.7760001 24,29.5000001 C24,29.2240001 24.2240001,29.0000001 24.5,29.0000001 L32.5,29.0000001 Z M22.853,16.147 C23.048,16.342 23.048,16.659 22.853,16.854 L17.853,21.854 C17.756,21.951 17.628,22 17.5,22 C17.372,22 17.244,21.951 17.146,21.854 L15.146,19.854 C14.951,19.659 14.951,19.342 15.146,19.147 C15.341,18.952 15.658,18.952 15.853,19.147 L17.5,20.793 L22.146,16.147 C22.341,15.952 22.658,15.952 22.853,16.147 Z M32.5,20 C32.7760001,20 33,20.224 33,20.5 C33,20.776 32.7760001,21 32.5,21 L24.5,21 C24.2240001,21 24,20.776 24,20.5 C24,20.224 24.2240001,20 24.5,20 L32.5,20 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
  <a
    className="lenkepanel innboks lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -366.000000)"
            >
              <g
                transform="translate(382.000000, 366.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M33,16 C34.1030001,16 35,16.897 35,18 L35,29 C35,30.1030001 34.1030001,31 33,31 L15,31 C13.897,31 13,30.1030001 13,29 L13,18 C13,16.897 13.897,16 15,16 L33,16 Z M14,18 L14,29 C14,29.5510001 14.448,30 15,30 L33,30 C33.5520001,30 34,29.5510001 34,29 L34,18 C34,17.8895814 33.9820088,17.7832589 33.9487949,17.6838252 L24.313,25.391 C24.221,25.463 24.11,25.5 24,25.5 C23.89,25.5 23.779,25.463 23.687,25.391 L14.0515345,17.682841 C14.0181101,17.7825572 14,17.8892169 14,18 Z M33,17 L15,17 C14.9396359,17 14.8805155,17.0053694 14.8230887,17.0156544 L24,24.36 L33.1798974,17.0161939 C33.1215346,17.0055572 33.0614109,17 33,17 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
  <a
    className="lenkepanel innboks lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -366.000000)"
            >
              <g
                transform="translate(382.000000, 366.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M33,16 C34.1030001,16 35,16.897 35,18 L35,29 C35,30.1030001 34.1030001,31 33,31 L15,31 C13.897,31 13,30.1030001 13,29 L13,18 C13,16.897 13.897,16 15,16 L33,16 Z M14,18 L14,29 C14,29.5510001 14.448,30 15,30 L33,30 C33.5520001,30 34,29.5510001 34,29 L34,18 C34,17.8895814 33.9820088,17.7832589 33.9487949,17.6838252 L24.313,25.391 C24.221,25.463 24.11,25.5 24,25.5 C23.89,25.5 23.779,25.463 23.687,25.391 L14.0515345,17.682841 C14.0181101,17.7825572 14,17.8892169 14,18 Z M33,17 L15,17 C14.9396359,17 14.8805155,17.0053694 14.8230887,17.0156544 L24,24.36 L33.1798974,17.0161939 C33.1215346,17.0055572 33.0614109,17 33,17 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
]
`;

exports[`Brukernotifikasjoner with beskjeder, oppgaver and innbokser, but no innlogging 1`] = `null`;

exports[`Brukernotifikasjoner with innbokser 1`] = `
Array [
  <a
    className="lenkepanel innboks lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -366.000000)"
            >
              <g
                transform="translate(382.000000, 366.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M33,16 C34.1030001,16 35,16.897 35,18 L35,29 C35,30.1030001 34.1030001,31 33,31 L15,31 C13.897,31 13,30.1030001 13,29 L13,18 C13,16.897 13.897,16 15,16 L33,16 Z M14,18 L14,29 C14,29.5510001 14.448,30 15,30 L33,30 C33.5520001,30 34,29.5510001 34,29 L34,18 C34,17.8895814 33.9820088,17.7832589 33.9487949,17.6838252 L24.313,25.391 C24.221,25.463 24.11,25.5 24,25.5 C23.89,25.5 23.779,25.463 23.687,25.391 L14.0515345,17.682841 C14.0181101,17.7825572 14,17.8892169 14,18 Z M33,17 L15,17 C14.9396359,17 14.8805155,17.0053694 14.8230887,17.0156544 L24,24.36 L33.1798974,17.0161939 C33.1215346,17.0055572 33.0614109,17 33,17 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
  <a
    className="lenkepanel innboks lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -366.000000)"
            >
              <g
                transform="translate(382.000000, 366.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M33,16 C34.1030001,16 35,16.897 35,18 L35,29 C35,30.1030001 34.1030001,31 33,31 L15,31 C13.897,31 13,30.1030001 13,29 L13,18 C13,16.897 13.897,16 15,16 L33,16 Z M14,18 L14,29 C14,29.5510001 14.448,30 15,30 L33,30 C33.5520001,30 34,29.5510001 34,29 L34,18 C34,17.8895814 33.9820088,17.7832589 33.9487949,17.6838252 L24.313,25.391 C24.221,25.463 24.11,25.5 24,25.5 C23.89,25.5 23.779,25.463 23.687,25.391 L14.0515345,17.682841 C14.0181101,17.7825572 14,17.8892169 14,18 Z M33,17 L15,17 C14.9396359,17 14.8805155,17.0053694 14.8230887,17.0156544 L24,24.36 L33.1798974,17.0161939 C33.1215346,17.0055572 33.0614109,17 33,17 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
]
`;

exports[`Brukernotifikasjoner with oppgaver 1`] = `
Array [
  <a
    className="lenkepanel oppgave lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -429.000000)"
            >
              <g
                transform="translate(382.000000, 429.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M35.5,12 C35.7760001,12 36,12.224 36,12.5 L36,35.5 C36,35.7760001 35.7760001,36 35.5,36 L12.5,36 C12.224,36 12,35.7760001 12,35.5 L12,12.5 C12,12.224 12.224,12 12.5,12 L35.5,12 Z M35,13 L13,13 L13,35 L35,35 L35,13 Z M22.853,25.1470001 C23.048,25.3420001 23.048,25.6590001 22.853,25.8540001 L17.853,30.8540001 C17.756,30.9510001 17.628,31.0000001 17.5,31.0000001 C17.372,31.0000001 17.244,30.9510001 17.146,30.8540001 L15.146,28.8540001 C14.951,28.6590001 14.951,28.3420001 15.146,28.1470001 C15.341,27.9520001 15.658,27.9520001 15.853,28.1470001 L17.5,29.7930001 L22.146,25.1470001 C22.341,24.9520001 22.658,24.9520001 22.853,25.1470001 Z M32.5,29.0000001 C32.7760001,29.0000001 33,29.2240001 33,29.5000001 C33,29.7760001 32.7760001,30.0000001 32.5,30.0000001 L24.5,30.0000001 C24.2240001,30.0000001 24,29.7760001 24,29.5000001 C24,29.2240001 24.2240001,29.0000001 24.5,29.0000001 L32.5,29.0000001 Z M22.853,16.147 C23.048,16.342 23.048,16.659 22.853,16.854 L17.853,21.854 C17.756,21.951 17.628,22 17.5,22 C17.372,22 17.244,21.951 17.146,21.854 L15.146,19.854 C14.951,19.659 14.951,19.342 15.146,19.147 C15.341,18.952 15.658,18.952 15.853,19.147 L17.5,20.793 L22.146,16.147 C22.341,15.952 22.658,15.952 22.853,16.147 Z M32.5,20 C32.7760001,20 33,20.224 33,20.5 C33,20.776 32.7760001,21 32.5,21 L24.5,21 C24.2240001,21 24,20.776 24,20.5 C24,20.224 24.2240001,20 24.5,20 L32.5,20 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
  <a
    className="lenkepanel oppgave lenkepanel--border"
    href="https://enNyLenke"
    onClick={null}
  >
    <div
      className="lenkepanel__innhold"
    >
      <div
        className="lenkepanel__ikon"
      >
        <svg
          height="48px"
          version="1.1"
          viewBox="0 0 48 48"
          width="48px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <g
              transform="translate(-382.000000, -429.000000)"
            >
              <g
                transform="translate(382.000000, 429.000000)"
              >
                <circle
                  cx="24"
                  cy="24"
                  fill="#FFBD66"
                  r="24"
                />
                <path
                  d="M35.5,12 C35.7760001,12 36,12.224 36,12.5 L36,35.5 C36,35.7760001 35.7760001,36 35.5,36 L12.5,36 C12.224,36 12,35.7760001 12,35.5 L12,12.5 C12,12.224 12.224,12 12.5,12 L35.5,12 Z M35,13 L13,13 L13,35 L35,35 L35,13 Z M22.853,25.1470001 C23.048,25.3420001 23.048,25.6590001 22.853,25.8540001 L17.853,30.8540001 C17.756,30.9510001 17.628,31.0000001 17.5,31.0000001 C17.372,31.0000001 17.244,30.9510001 17.146,30.8540001 L15.146,28.8540001 C14.951,28.6590001 14.951,28.3420001 15.146,28.1470001 C15.341,27.9520001 15.658,27.9520001 15.853,28.1470001 L17.5,29.7930001 L22.146,25.1470001 C22.341,24.9520001 22.658,24.9520001 22.853,25.1470001 Z M32.5,29.0000001 C32.7760001,29.0000001 33,29.2240001 33,29.5000001 C33,29.7760001 32.7760001,30.0000001 32.5,30.0000001 L24.5,30.0000001 C24.2240001,30.0000001 24,29.7760001 24,29.5000001 C24,29.2240001 24.2240001,29.0000001 24.5,29.0000001 L32.5,29.0000001 Z M22.853,16.147 C23.048,16.342 23.048,16.659 22.853,16.854 L17.853,21.854 C17.756,21.951 17.628,22 17.5,22 C17.372,22 17.244,21.951 17.146,21.854 L15.146,19.854 C14.951,19.659 14.951,19.342 15.146,19.147 C15.341,18.952 15.658,18.952 15.853,19.147 L17.5,20.793 L22.146,16.147 C22.341,15.952 22.658,15.952 22.853,16.147 Z M32.5,20 C32.7760001,20 33,20.224 33,20.5 C33,20.776 32.7760001,21 32.5,21 L24.5,21 C24.2240001,21 24,20.776 24,20.5 C24,20.224 24.2240001,20 24.5,20 L32.5,20 Z"
                  fill="#3E3832"
                  id="Combined-Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div>
        <p
          className="typo-element lenkepanel__heading"
        >
          Du har en sykemelding som m책 godkjennes
        </p>
        
        <p
          className="typo-undertekst lenkepanel__etikett"
        >
          13.03.2020
        </p>
      </div>
    </div>
    <span
      className="lenkepanel__indikator"
    />
  </a>,
]
`;
